app/src/main/java/org/sagebionetworks/research/mpower/tracking/fragment/MedicationDayFragment.kt

```diff
@@ -54,27 +54,27 @@ class MedicationDayFragment : AppCompatDialogFragment() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
 
-        var days : String? = null
+        var days: String? = null
         if (savedInstanceState == null) {
             if (arguments != null) {
-                schedId = arguments!!.getString(ARG_SCHED_ID)
-                name = arguments!!.getString(ARG_NAME)
-                time = arguments!!.getString(ARG_TIME)
+                arguments!!.getString(ARG_SCHED_ID)?.let { schedId = "" }
+                arguments!!.getString(ARG_NAME)?.let { name = "" }
+                arguments!!.getString(ARG_TIME)?.let { time = "" }
                 days = arguments!!.getString(ARG_DAYS)
             } else {
                 LOGGER.warn("No arguments found")
+                schedId = ""
                 name = "Default"
                 time = "9:15 PM"
             }
         } else {
-            schedId = savedInstanceState.getString(ARG_SCHED_ID)
-            name = savedInstanceState.getString(ARG_NAME)
-            time = savedInstanceState.getString(ARG_TIME)
+            savedInstanceState.getString(ARG_SCHED_ID)?.let { schedId = "" }
```

Comment:
You're using it opposite.  The let function only gets called if the value is non-null.  So you would want to do this...

savedInstanceState.getString(ARG_SCHED_ID)?.let { schedId = it } ?: run { schedId = "" }